src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java

```diff
@@ -351,6 +355,14 @@ public void execute() throws MojoExecutionException {
                     throw new MojoExecutionException("You have to use a classifier "
                             + "to attach supplemental artifacts to the project instead of replacing them.");
                 }
+
+                if (!"java"
+                        .equals(artifactHandlerManager
+                                .getArtifactHandler(getProject().getPackaging())
+                                .getLanguage())) {
+                    getLog().warn(
+                                    "The project packaging language is NOT 'java'; this is most probably not what you want");
```

Comment:
What do you think to add a packaging name ... like:
`The project packaging <packaging> language ...`